src/main/kotlin/lgp/core/evolution/ResultsExporter.kt

```diff
@@ -0,0 +1,339 @@
+                    }
+                }
+            }
+        }
+
+        private fun log(message: String) {
+            if (this.verbose)
+                println(message)
+        }
+
+        /**
+         * Provides information about the module.
+         */
+        override val information = ModuleInformation(
+                description = "Aggregates results that are able to be exported."
+        )
+    }
+
+    /**
+     * A [ResultAggregator] implementation which does nothing.
+     *
+     * Useful for situations where it is not required to aggregate results and export to an output source.
+     */
+    class DefaultResultAggregator<T> : ResultAggregator<T>() {
+        override val results = emptyList<ExportableResult<T>>()
+
+        override fun add(result: ExportableResult<T>) {
+            // No op.
+        }
+
+        override fun addAll(results: List<ExportableResult<T>>) {
+            // No op.
+        }
+
+        override fun close() {
+            // No op.
+        }
+
+        override val information = ModuleInformation(
+            "Default aggregator which simply does nothing."
+        )
+    }
+}
+
+/**
+ * Provides the ability to output results from a [ResultAggregator].
+ */
+interface ResultOutputProvider<T> {
+
+    /**
+     * Writes any results from [source] to some output destination.
+     */
+    fun writeResultsFrom(source: ResultAggregator<T>)
+}
+
+/**
+ * A collection of [ResultOutputProvider] implementations for common scenarios.
+ */
+object ResultOutputProviders {
+
+    /**
+     * Provides the ability to write a table of results to a text file.
+     *
+     * @param filename A file to write text results to.
+     */
+    class RawResultTableOutputProvider<T>(private val filename: String): ResultOutputProvider<T> {
+
+        private val writer = Files.newBufferedWriter(Paths.get(this.filename))
+
+        override fun writeResultsFrom(source: ResultAggregator<T>) {
+            writer.use { out ->
+                out.write(source.toString())
+            }
+        }
+    }
+
+    /**
+     * Provides the ability to write results to a CSV file.
+     *
+     * @param filename A file to write CSV data to.
+     */
+    class CsvResultOutputProvider<T>(private val filename: String) : ResultOutputProvider<T> {
+
+        private val csvWriter: CSVWriter = CSVWriter(
+                Files.newBufferedWriter(Paths.get(this.filename))
+        )
+
+        private var headerHasBeenWriten: Boolean = false
+
+        override fun writeResultsFrom(source: ResultAggregator<T>) {
+            // Act on a view of the results to avoid thread-safety issues
+            val results = source.results.toList()
+
+            // Is there any results?
+            if (results.isEmpty())
+                return
+
+            // Grab header information
+            if (!headerHasBeenWriten) {
+                val first = results.first().export()
+                val header = first.map { (name, _) -> name }
```

Comment:
Similar to other comment -- could change to:

```
val header = this.results.first().export().map(Pair::first)
```

Alternatively, because the syntax used here is more descriptive we could adopt it in the previous comment.